dashboard/index.js

```diff
@@ -1,319 +1,356 @@
-/* eslint-disable */
 "use strict";
 
-var blessed = require("blessed");
+const blessed = require("blessed");
 
-var formatOutput = require("../utils/format-output.js");
-var formatModules = require("../utils/format-modules.js");
-var formatAssets = require("../utils/format-assets.js");
+const formatOutput = require("../utils/format-output.js");
+const formatModules = require("../utils/format-modules.js");
+const formatAssets = require("../utils/format-assets.js");
 
-function Dashboard(options) {
-  var title = options && options.title || "webpack-dashboard";
+const PERCENT_MULTIPLIER = 100;
 
-  this.color = options && options.color || "green";
-  this.minimal = options && options.minimal || false;
-  this.setData = this.setData.bind(this);
+class Dashboard {
+  constructor(options) {
+    const title = options && options.title || "webpack-dashboard";
 
-  this.screen = blessed.screen({
-    smartCSR: true,
-    title: title,
-    dockBorders: false,
-    fullUnicode: true,
-    autoPadding: true
-  });
+    this.color = options && options.color || "green";
+    this.minimal = options && options.minimal || false;
+    this.setData = this.setData.bind(this);
 
-  this.layoutLog.call(this);
-  this.layoutStatus.call(this);
-  !this.minimal && this.layoutModules.call(this);
-  !this.minimal && this.layoutAssets.call(this);
+    this.screen = blessed.screen({
+      title,
+      smartCSR: true,
+      dockBorders: false,
+      fullUnicode: true,
+      autoPadding: true
+    });
 
-  this.screen.key(["escape", "q", "C-c"], function() {
-    process.exit(0);
-  });
+    this.layoutLog();
+    this.layoutStatus();
+    if (!this.minimal) {
+      this.layoutModules();
+      this.layoutAssets();
+    }
 
-  this.screen.render();
-}
+    this.screen.key(["escape", "q", "C-c"], () => {
+      // eslint-disable-next-line no-process-exit
+      process.exit(0);
+    });
 
-Dashboard.prototype.setData = function(dataArr) {
-  var self = this;
-
-  dataArr.forEach(function(data) {
-    switch (data.type) {
-      case "progress": {
-        var percent = parseInt(data.value * 100);
-        if (self.minimal) {
-          percent && self.progress.setContent(percent.toString() + "%");
-        } else {
-          percent && self.progressbar.setContent(percent.toString() + "%");
-          self.progressbar.setProgress(percent);
-        }
+    this.screen.render();
+  }
+
+  setData(dataArray) {
+    dataArray.forEach(data => {
+      switch (data.type) {
```

Comment:
I broke out this switch statement into separate methods since there'll be more LOC here in the next PR.